### PR TITLE
Remove Helper Method Frames from debugdebugger

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Each commit contains a single functions from the diagnostic space. This PR is best reviewed by each commit.

Removed unused JIT Helper for x86.

Ran diagnostics tests and confirmed no failures.

### CODE REVIEW

File: src/coreclr/vm/debugdebugger.cpp [hunk 1]
```diff
@@ -230,43 +226,221 @@ FCIMPL0(FC_BOOL_RET, DebugDebugger::IsLogging)
 }
 FCIMPLEND
 
-FCIMPL4(void, DebugStackTrace::GetStackFramesInternal,
-        StackFrameHelper* pStackFrameHelperUNSAFE,
-        INT32 iSkip,
-        FC_BOOL_ARG fNeedFileInfo,
-        Object* pExceptionUNSAFE
-       )
+static StackWalkAction GetStackFramesCallback(CrawlFrame* pCf, VOID* data)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/debugdebugger.cpp:229]: Converted from member function to private in translation unit. This was moved from below.

File: src/coreclr/vm/debugdebugger.cpp [hunk 2]
```diff
@@ -230,43 +226,221 @@ FCIMPL0(FC_BOOL_RET, DebugDebugger::IsLogging)
...
+    // check if we already have the number of frames that the user had asked for
+    if ((pData->NumFramesRequested != 0) && (pData->NumFramesRequested <= pData->cElements))
+    {
+        return SWA_ABORT;
+    }
+
+    return SWA_CONTINUE;
+}
+
+static void GetStackFrames(
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/debugdebugger.cpp:313]: Converted from member function to private in translation unit. This was moved from below.

File: src/coreclr/vm/debugdebugger.cpp [hunk 3]
```diff
@@ -770,242 +943,34 @@ typedef Wrapper<OBJECTHANDLE, DoNothing<OBJECTHANDLE>, HolderDestroyStrongHandle
...
-
-    OBJECTREF pData = ObjectToOBJECTREF(dataUNSAFE);
+    QCALL_CONTRACT;
 
 #ifdef DEBUGGING_SUPPORTED
     // Send notification only if the debugger is attached
-    if (CORDebuggerAttached() )
-    {
-        HELPER_METHOD_FRAME_BEGIN_PROTECT(pData);
+    _ASSERTE(CORDebuggerAttached());
```
Comment (jkotas) [inline src/coreclr/vm/debugdebugger.cpp:952]: Nit: There is a small chance that the debugger may detach between the check on the managed side and here.

File: src/coreclr/vm/debugdebugger.cpp [hunk 4]
```diff
@@ -230,43 +226,221 @@ FCIMPL0(FC_BOOL_RET, DebugDebugger::IsLogging)
...
+
+    if (pData->TargetThread == NULL ||
+        pData->TargetThread->GetInternal() == GetThread())
+    {
+        // Null target thread specifies current thread.
+        GetThread()->StackWalkFrames(GetStackFramesCallback, pData, FUNCTIONSONLY | QUICKUNWIND, pStartFrame);
+    }
+    else
+    {
+        Thread *pThread = pData->TargetThread->GetInternal();
```
Comment (jkotas) [inline src/coreclr/vm/debugdebugger.cpp:352]: This path looks like a dead code. As far as I can tell, `TargetThread` is always set to null.

File: src/coreclr/vm/debugdebugger.cpp [hunk 5]
```diff
@@ -230,43 +226,221 @@ FCIMPL0(FC_BOOL_RET, DebugDebugger::IsLogging)
...
+    {
+        return SWA_ABORT;
+    }
+
+    return SWA_CONTINUE;
+}
+
+static void GetStackFrames(
+    Frame *pStartFrame,
+    void* pStopStack,
```
Comment (jkotas) [inline src/coreclr/vm/debugdebugger.cpp:315]:
```suggestion
```
These two arguments are unnecessary - the values can be inlined in the implementation.

File: src/coreclr/vm/debugdebugger.cpp (hunk 3 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/debugdebugger.cpp:952]: I will re-enable the check.

File: src/coreclr/vm/debugdebugger.cpp (hunk 4 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/debugdebugger.cpp:352]: Agreed. I will clean this up.

File: src/coreclr/vm/debugdebugger.cpp [hunk 6]
```diff
@@ -230,43 +226,221 @@ FCIMPL0(FC_BOOL_RET, DebugDebugger::IsLogging)
...
+    gc.dynamicMethodArrayOrig = NULL;
+    GCPROTECT_BEGIN(gc);
+    gc.pStackFrameHelper = (STACKFRAMEHELPERREF)stackFrameHelper.Get();
+    gc.pException = exception.Get();
+
+    DebugStackTrace::GetStackFramesData data;
 
     data.pDomain = GetAppDomain();
 
     data.skip = iSkip;
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/debugdebugger.cpp:436]: it works out just fine. In fact, we always seem to be passing 0.

**Comment by AaronRobinsonMSFT** [comment]
@dotnet/dotnet-diag and @jkotas All green.

**Review by jkotas — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**